Bump Stack to 3.7.1

waivio opened the pull request:
The latest nightly for `ghc-9.12.2` needs a stack upgrade to stack-3.7.1.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

This time when I ran `pkgcheck scan --commits --net` it said:
`pkgcheck: error: repos.conf: default repo 'gentoo' is undefined or invalid`

That didn't look right, but I made recent changes to my overlays, so I'll probably have to dig into that further.  Everything was made with a `hackport merge <blah>` command and seems to work right when tested.  No attempt at disabling network tests, probably should be done by someone who knows better.

Thanks,
NW

waivio commented:
Ok, I think I fixed the issue with pkgcheck on my system.  The output is now:
```
dev-haskell/open-browser
  RedundantVersion: version 0.2.1.0-r3: slot(0) keywords are overshadowed by version: 0.4.0.0

dev-haskell/unix-compat
  UnusedLocalUse: unused local USE flag: [ old-time ]
  RedundantVersion: version 0.7.3: slot(0) keywords are overshadowed by version: 0.7.4

dev-haskell/crypton
  RedundantVersion: version 1.0.1: slot(0) keywords are overshadowed by version: 1.0.4
  BadDefaultUseFlags: version 1.0.4: bad default USE flags: '+cpu_flags_x86_aes', '+cpu_flags_x86_pclmul', '+cpu_flags_x86_rdrand'

dev-haskell/conduit
  RedundantVersion: version 1.3.6: slot(0) keywords are overshadowed by version: 1.3.6.1

dev-haskell/mustache
  RedundantVersion: version 2.4.2: slot(0) keywords are overshadowed by version: 2.4.3.1

dev-haskell/pantry
  RedundantVersion: version 0.10.0: slot(0) keywords are overshadowed by version: 0.11.0

dev-haskell/stack
  ExcessiveLineLength: version 3.1.1: excessive line length (over 120 characters) on line: 7
  RedundantVersion: version 3.1.1: slot(0) keywords are overshadowed by version: 3.7.1
  ExcessiveLineLength: version 3.7.1: excessive line length (over 120 characters) on line: 7
  NonexistentDeps: version 3.7.1: DEPEND: nonexistent package: >=dev-haskell/hspec-2.11.12
  NonsolvableDepsInDev: version 3.7.1: nonsolvable depset(depend) keyword(~amd64) dev profile (default/linux/amd64/23.0/musl) (6 total): solutions: [ >=dev-haskell/hspec-2.11.12 ]
  NonsolvableDepsInStable: version 3.7.1: nonsolvable depset(depend) keyword(~amd64) stable profile (default/linux/amd64/23.0) (56 total): solutions: [ >=dev-haskell/hspec-2.11.12 ]
  ExcessiveLineLength: version 9.9.9: excessive line length (over 120 characters) on line: 7
```

I think I need to add `dev-haskell/hspec-2.11.12` for some of the test suite deps.  I'll try and get that added soon.

Thanks,
NW

GuillaumedeVolpiano commented on dev-haskell/stack/stack-3.7.1.ebuild:
```diff
@@ -0,0 +1,89 @@
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="disable-stack-upload"
+
```
You need to add back the patch from 3.1.1 here. network tests should be disabled on Gentoo for security reasons.

`PATCHES=(
    "${FILESDIR}/${PN}-2.13.1-disable-network-tests.patch"
)`

waivio replied:
Added back the patch to disable network tests.

GuillaumedeVolpiano reviewed (changes requested):
You should drop conduit-1.3.6 also.

GuillaumedeVolpiano commented on dev-haskell/crypton/crypton-1.0.4.ebuild:
```diff
@@ -0,0 +1,43 @@
+		dev-haskell/tasty-kat
+		dev-haskell/tasty-quickcheck )
+"
+
```
Isn't this patch still needed?

```
PATCHES=(
    "${FILESDIR}"/${P}-fix-tests.patch
    )
```

waivio replied:
It looks like the issue was solved upstream in https://github.com/kazu-yamamoto/crypton/issues/40

GuillaumedeVolpiano commented on dev-haskell/mustache/mustache-2.4.3.1.ebuild:
```diff
@@ -0,0 +1,38 @@
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+
```
Isn't this patch still needed too?

`PATCHES=(
    "${FILESDIR}/${PN}-2.3.2-disable-langspec-tests.patch"
)`

waivio replied:
This patch is still needed added commit to disable the language-spec tests.

GuillaumedeVolpiano commented on dev-haskell/pantry/pantry-0.11.0.ebuild:
Please drop pantry-0.10.0

GuillaumedeVolpiano commented on dev-haskell/pantry/pantry-0.11.0.ebuild:
```diff
@@ -0,0 +1,75 @@
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+
```
Add patch back

`PATCHES=(
    "${FILESDIR}/${PN}-0.9.3.2-disable-network-tests.patch"
)`

waivio replied:
Added patch back to pantry-0.11.0.ebuild

GuillaumedeVolpiano commented on dev-haskell/unix-compat/unix-compat-0.7.4.ebuild:
Please drop unix-compat-0.7.3

GuillaumedeVolpiano commented on dev-haskell/hspec/hspec-2.11.12.ebuild:
Please drop hspec-2.11.10. You'll need to relax the dependencies or bump the version on toml-parser as a consequence.

waivio replied:
dropped hspec-2.11.10, toml-parser seems to support it...
the toml-parser ebuild has this:
```
	test? ( || ( ( >=dev-haskell/hspec-2.10 <dev-haskell/hspec-2.11 )
			( >=dev-haskell/hspec-2.11 <dev-haskell/hspec-2.12 ) ) )
```
So I believe toml-parser does support hspec-2.11.12

GuillaumedeVolpiano commented on dev-haskell/hspec-core/hspec-core-2.11.12.ebuild:
Please drop hspec-core-2.11.10. You'll also need to bump the version of hspec-api.

waivio commented on dev-haskell/crypton/crypton-1.0.4.ebuild:
dropped crypton-1.0.1

waivio commented:
I wanted to get some of the dropped package changes up.  Still got more to do...

pkgcheck scan gave me this warning:
```
dev-haskell/crypton
  BadDefaultUseFlags: version 1.0.4: bad default USE flags: '+cpu_flags_x86_aes', '+cpu_flags_x86_pclmul', '+cpu_flags_x86_rdrand'
```
I think it should be good, because the previous package seemed to use the specific flags.  Who'd want to slow down crypto type stuff?

I'm going to have to dig in deeper to the changes to the tests to figure out what to do, and see if up-stream has an open issue for the tests.

Also need to dig into the `open-bowser` can of worms and eat them one-by-one.

I think I also f'd up my commit message for `unix-compat`, I hope to fix that next.

Nathan.

waivio commented on dev-haskell/open-browser/open-browser-0.4.0.0.ebuild:
cabal-install on Hackage has a revision to loosen the deps, so this should be pretty easy to change.
pandoc-crossref has an open issue upstream to loosen the dep: https://github.com/lierdakil/pandoc-crossref/issues/492

@GuillaumedeVolpiano , should the cabal-install change happen in a separate PR or as part of this one?

waivio commented:
Fixed commit messages and squashed some commits for easier review.  I think the only open TODO is loosening `open-browser` dep  for `pandoc-crossref`.
